pkg/infrastructure/platform/platform.go

```diff
@@ -45,10 +45,7 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 	case awstypes.Name:
 		return clusterapi.InitializeProvider(&awscapi.Provider{}), nil
 	case azuretypes.Name:
-		if types.ClusterAPIFeatureGateEnabled(platform, fg) {
```

Comment:
you can drop this change. It is handled in #8809

Reply:
I can't change it back since I've removed terraform in this PR. This will need to wait for #8809 to merge.